pkg/serviceaccount/claims.go

```diff
@@ -155,7 +155,7 @@ func (v *validator) Validate(_ string, public *jwt.Claims, privateObj interface{
 			glog.V(4).Infof("Bound secret is deleted and awaiting removal: %s/%s for service account %s/%s", namespace, secref.Name, namespace, saref.Name)
 			return "", "", "", errors.New("Token has been invalidated")
 		}
-		if string(secref.UID) != secref.UID {
+		if string(secref.UID) != secret.UID {
 			glog.V(4).Infof("Secret UID no longer matches %s/%s: %q != %q", namespace, secref.Name, string(serviceAccount.UID), secref.UID)
```

Comment:
Looks like we should also update the log info:
string(serviceAccount.UID) -> secret.UID

Reply:
done